### PR TITLE
Adapt to version 377

### PR DESCRIPTION
**Pull request by maltesander**
Adapted the trino-opa-authorizer to the new or extended interface of Trino 377.

We now send all the information we get to OPA and use only one rule `allow`. The hard work must then be done in the rego rules.

To test this you can setup a local Trino instance (download and extract version 377). Copy the extracted [etc.tar.gz](https://github.com/stackabletech/trino-opa-authorizer/files/8514852/etc.tar.gz) in the `${TRINO_HOME}/etc` folder (the required config files).

Follow the instructions in the readme (build and copy the `target/out` files into the `${TRINO_HOME}/plugin/trino-opa-authorizer`

Download and run OPA (27.1 does not work with the example.rego, use 37.1) with the example provided in the repo (you need it available for the mvn test or mvn package commands as well:
`./opa run --log-level debug --server ./example.rego `

Run Trino:
`${TRINO_HOME}/bin/launcher run`

Use the [CLI tool](https://repo.stackable.tech/repository/packages/trino-cli/trino-cli-377-executable.jar) to test. The user admin should be able do to everything while user bob is restricted to see only catalogs and run some queries:
```
./trino-cli-377-executable.jar --server http://localhost:8080 --user=bob
```

and try:

```
trino> SHOW CATALOGS;
 Catalog
---------
 system
(1 row)

Query 20220419_161711_00047_bu682, FINISHED, 1 node
Splits: 1 total, 1 done (100.00%)
0.22 [0 rows, 0B] [0 rows/s, 0B/s]
```
and
```
trino> SHOW SCHEMAS From system;
Query 20220419_161650_00046_bu682 failed: Access Denied: Cannot show schemas
```

part of https://github.com/stackabletech/trino-operator/issues/187

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -20,3 +19,63 @@ Assuming that you already have a Trino server set up.
 === Tests
 
 The tests assume that OPA is available on your path as `opa`, and can then be executed by running `mvn test`.
+
+=== Example
+
+The example `example.rego` allows the `admin` user to do everything. User `bob` is limited to run queries and access catalogs.
+
+----
```
Comment (sbernauer) [inline README.adoc:27]: Having an include of `example.rego` would be great

File: README.adoc [hunk 2]
```diff
@@ -20,3 +19,63 @@ Assuming that you already have a Trino server set up.
...
+
+Query 20220419_161711_00047_bu682, FINISHED, 1 node
+Splits: 1 total, 1 done (100.00%)
+0.22 [0 rows, 0B] [0 rows/s, 0B/s]
+
+----
+but should get an `Access Denied` failure when e.g. showing schemas:
+----
+trino> SHOW SCHEMAS From system;
+Query 20220419_161650_00046_bu682 failed: Access Denied: Cannot show schemas
```
Comment (sbernauer) [inline README.adoc:81]:
```suggestion
Query 20220419_161650_00046_bu682 failed: Access Denied: Cannot show schemas
----
```

File: pom.xml [hunk 3]
```diff
@@ -6,7 +6,7 @@
 
     <groupId>tech.stackable.trino</groupId>
     <artifactId>trino-opa-authorizer</artifactId>
-    <version>0.1.0-SNAPSHOT</version>
+    <version>377</version>
```
Comment (sbernauer) [inline pom.xml:9]: I guess good decision as the interface seems to change quite often if i understand you correctly

**Review by sbernauer — COMMENTED** [review]
Haven't looked at the implementation (yet?) as i think other people worked already on it and know better. Just some minor doc comments

File: README.adoc (hunk 1 above)
Comment (maltesander) [reply README.adoc:27]: done

File: README.adoc (hunk 2 above)
Comment (maltesander) [reply README.adoc]: done

File: pom.xml (hunk 3 above)
Comment (maltesander) [reply pom.xml:9]: Yeah i normally do not like that but i think in this case we need to release / adapt to the trino changes alot. It maybe would make sense to still keep some actual product version like 0.1.0 to have some versioning.

**Review by razvan — APPROVED** [review]
Successfully followed the test instructions

**Pull request merged**